src/main/java/hudson/plugins/analysis/collector/AnalysisPublisher.java

```diff
@@ -31,113 +40,23 @@
-     * @param isOpenTasksActivated
-     *            determines whether to collect open tasks
-     * @param isWarningsActivated
-     *            determines whether to collect compiler warnings
-     * @param canRunOnFailed
-     *            determines whether the plug-in can run for failed builds, too
-     * @param usePreviousBuildAsReference
-     *            determines whether the previous build should be used as the
-     *            reference build
-     * @param useStableBuildAsReference
-     *            determines whether only stable builds should be used as reference builds or not
-     * @param canComputeNew
-     *            determines whether new warnings should be computed (with
-     *            respect to baseline)
+     * Default constructor.
+     * Use setter to initialize fields if needed.
      */
-    // CHECKSTYLE:OFF
-    @SuppressWarnings("PMD.ExcessiveParameterList")
     @DataBoundConstructor
-    public AnalysisPublisher(final String healthy, final String unHealthy, final String thresholdLimit,
-            final String defaultEncoding, final boolean useDeltaValues,
-            final String unstableTotalAll, final String unstableTotalHigh, final String unstableTotalNormal, final String unstableTotalLow,
-            final String unstableNewAll, final String unstableNewHigh, final String unstableNewNormal, final String unstableNewLow,
-            final String failedTotalAll, final String failedTotalHigh, final String failedTotalNormal, final String failedTotalLow,
-            final String failedNewAll, final String failedNewHigh, final String failedNewNormal, final String failedNewLow,
-            final boolean isCheckStyleActivated, final boolean isDryActivated, final boolean isFindBugsActivated,
-            final boolean isPmdActivated, final boolean isOpenTasksActivated, final boolean isWarningsActivated,
-            final boolean canRunOnFailed,
-            final boolean usePreviousBuildAsReference, final boolean useStableBuildAsReference, final boolean canComputeNew) {
-        super(healthy, unHealthy, thresholdLimit, defaultEncoding, useDeltaValues,
-                unstableTotalAll, unstableTotalHigh, unstableTotalNormal, unstableTotalLow,
-                unstableNewAll, unstableNewHigh, unstableNewNormal, unstableNewLow,
-                failedTotalAll, failedTotalHigh, failedTotalNormal, failedTotalLow,
-                failedNewAll, failedNewHigh, failedNewNormal, failedNewLow,
-                canRunOnFailed, usePreviousBuildAsReference, useStableBuildAsReference, false, canComputeNew, false, "ANALYSIS-COLLECTOR");
-        isDryDeactivated = !isDryActivated;
-        isFindBugsDeactivated = !isFindBugsActivated;
-        isPmdDeactivated = !isPmdActivated;
-        isOpenTasksDeactivated = !isOpenTasksActivated;
-        isWarningsDeactivated = !isWarningsActivated;
-        isCheckStyleDeactivated = !isCheckStyleActivated;
+    public AnalysisPublisher() {
```

Comment:
See [this comment](https://github.com/jenkinsci/analysis-collector-plugin/pull/2#discussion_r37967954)